src/pp_roaming_protocol.erl

```diff
@@ -362,7 +371,8 @@ handle_xmitdata_req(#{
         {error, _} = Err ->
             Err;
         {ok, TransactionID, Region, PacketTime, DestURL, FlowType} ->
-            DataRate = pp_lorawan:index_to_datarate(Region, DR),
+            Plan = lora_plan:region_to_plan(Region),
+            DataRate = lora_plan:datarate_to_string(Plan, DR),
```

Comment:
This pair looks like an idiom. Maybe create a utility function to convert a Region to DataRate and hide the Plan?

Reply:
This is what the `pp_lorawan` module is for.